core/lib/state/src/rocksdb/mod.rs

```diff
@@ -8,18 +8,22 @@
 //! - Contracts
 //! - Factory dependencies
 //!
-//! | Column       | Key                    | Value                   | Description                          |
-//! | ------------ | ---------------------- | ----------------------- | ------------------------------------ |
-//! | State        | 'block_number'         | serialized block number | Last processed L1 batch number (u32) |
-//! | State        | hashed `StorageKey`    | 32 bytes value          | State for the given key              |
-//! | Contracts    | address (20 bytes)     | `Vec<u8>`               | Contract contents                    |
-//! | Factory deps | hash (32 bytes)        | `Vec<u8>`               | Bytecodes for new contracts that a certain contract may deploy. |
-
-use std::{collections::HashMap, mem, path::Path, time::Instant};
+//! | Column       | Key                             | Value                           | Description                               |
+//! | ------------ | ------------------------------- | ------------------------------- | ----------------------------------------- |
+//! | State        | 'block_number'                  | serialized block number         | Last processed L1 batch number (u32)      |
+//! | State        | 'enum_index_migration_progress' | serialized next key to process  | Hashed key migration should continue from |
+//! | State        | 'enum_index_migration_finished' | 1 byte flag                     | Denotes if the migration if completed     |
+//! | State        | hashed `StorageKey`             | 32 bytes value ++ 8 bytes index | State for the given key                   |
```

Comment:
For specificity, you should specify index endianness. Also, why is it big-endian anyway? We don't need to sort by the index (which is where big-endian encoding would work and little-endian wouldn't). Not that it matters much, but most modern CPU architectures use little-endian. Also, if we care about space usage that much, we could use variable-length encoding (e.g., [LEB128](https://en.wikipedia.org/wiki/LEB128) used in Protobuf; it's used in the Merkle tree crate).

Reply:
I don't think we care that much about space usage at the moment. I wouldn't use LEB128 for indices now, probably in the future we can use it for both values and indices. Are there any benefits in using little-endianness without variable-length encoding?

Reply:
FWIW both little and big endianness in used in the VM  code - so whatever we chose here we won't be 100% consistent. I'd keep as is

Reply:
Well, keys and values are now stored in state keeper's RockDB in big-endian format, so I'd choose big-, unless @slowli has a strong reason to use little-

Reply:
No particularly strong opinion; I'm fine with leaving it as is. Regarding values, they *probably* should be interpreted as 32-byte arrays (`H256` in our terms) rather than unsigned integers with specific endianness (`U256` in our terms) because we don't really rely on any arithmetic operations performed on them on the storage level 🙃 (This is outside the scope of this PR, I'm pretty sure.) For keys, using big-endian encoding makes sense because it orders them in the natural order.